MapboxNavigation/NavigationMapView.swift

```diff
@@ -786,7 +792,14 @@ open class NavigationMapView: MGLMapView, UIGestureRecognizerDelegate {
     // MARK: Utility Methods
     
     func shape(describing route: Route, legIndex: Int?) -> MGLShape? {
-        guard let coordinates = route.coordinates else { return nil }
+        guard let routeCoords = route.coordinates else { return nil }
+        
+        let coordinates: [CLLocationCoordinate2D]
+        if let userPuckScreenCoordinates = userPuckScreenCoordinates {
```

Comment:
Handle cases where the view may not be using a `UserPuckView`.

Reply:
Just a wild thought, but why not use `NavigationMapViewDelegate.navigationMapViewUserAnchorPoint(_:)`? That saves us from having to do the screen coordinates conversion, and I think we already define default behavior if the developer doesn't implement the method?

Reply:
Just a wild thought, but why not use `NavigationMapViewDelegate.navigationMapViewUserAnchorPoint(_:)`? That saves us from having to do the screen coordinates conversion, and I think we already define default behavior if the developer doesn't implement the method?

Reply:
Unsure why, but when I used this delegate, it clipped the route line to the upcoming maneuver.